wagtail/models/__init__.py

```diff
@@ -649,6 +649,51 @@ def _get_response(self, request):
         handler.load_middleware()
         return handler.get_response(request)
 
+    def make_draft_request(
+        self, original_request=None, preview_mode=None, extra_request_attrs=None
+    ):
+        """
+        Simulate a request to this object, by constructing a fake HttpRequest object that is (as far
+        as possible) representative of a real request to this object's front-end URL, and invoking
+        serve_preview with that request (and the given preview_mode).
+
+        Used for previewing / moderation and any other place where we
+        want to display a view of this object in the admin interface without going through the regular
+        page routing logic.
+
+        If you pass in a real request object as original_request, additional information (e.g. client IP, cookies)
+        will be included in the dummy request.
+        """
+        dummy_meta = self._get_dummy_headers(original_request)
+        request = WSGIRequest(dummy_meta)
+
+        # Add a flag to let middleware know that this is a dummy request.
+        request.is_dummy = True
+
+        if extra_request_attrs:
+            for k, v in extra_request_attrs.items():
+                setattr(request, k, v)
+
+        obj = self
+
+        # Build a custom django.core.handlers.BaseHandler subclass that invokes serve_preview as
+        # the eventual view function called at the end of the middleware chain, rather than going
+        # through the URL resolver
+        class Handler(BaseHandler):
+            def _get_response(self, request):
+                request.is_preview = False
```

Comment:
Other than this line, this is an exact duplicate of `make_preview_request`. Setting this to `False` isn't correct either, since this technically *is* a preview.

I'd suggest we go with a simpler approach (that also addresses other similar issues) as outlined in https://github.com/wagtail/wagtail/issues/10002#issuecomment-1418920909.

Reply:
Okay I will make the changes accordingly as outlined in #10002.

Reply:
As per your suggestions I made the changes and now the following pages look like this
1. Preview on Edit/Create
<img width="960" alt="Screenshot 2023-02-07 235040" src="https://user-images.githubusercontent.com/96590873/217332593-c7d5ef23-79c1-4255-a224-7339b81bdc99.png">

2. Workflow Preview
<img width="960" alt="Screenshot 2023-02-07 234940" src="https://user-images.githubusercontent.com/96590873/217333022-3aa90add-ca63-4992-8de3-3e0e1427966d.png">

3. Revision Preview
<img width="960" alt="Screenshot 2023-02-07 234848" src="https://user-images.githubusercontent.com/96590873/217333109-226a87fa-0277-41cd-90e5-faf731f7bc43.png">

4. View Draft
<img width="960" alt="Screenshot 2023-02-07 234745" src="https://user-images.githubusercontent.com/96590873/217333164-938be63d-c80a-441a-8998-1ac23479423b.png">

Reply:
Hi @laymonage, I made the changes as you suggested and also modified the tests accordingly to check the edit button on view draft, revision preview and workflow preview. Could you please re review it and suggest any further changes?